File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/model/annotation/PdfAnnotationExtras.kt
package dev.subfly.yabacore.model.annotation

import kotlinx.serialization.Serializable

/** Serialized into [dev.subfly.yabacore.database.entities.AnnotationEntity.extrasJson] for [AnnotationType.PDF]. */
@Serializable
data class PdfAnnotationExtras(
    val startSectionKey: String,
    val startOffsetInSection: Int,
    val endSectionKey: String,
    val endOffsetInSection: Int,
)

File: Darwin/YABACore/Model/Annotation/PdfAnnotationExtras.swift
//
//  PdfAnnotationExtras.swift
//  YABACore
//
//  Parity with Compose `PdfAnnotationExtras`.
//

import Foundation

public struct PdfAnnotationExtras: Sendable, Codable, Equatable {
    public var startSectionKey: String
    public var startOffsetInSection: Int
    public var endSectionKey: String
    public var endOffsetInSection: Int

    public init(
        startSectionKey: String,
        startOffsetInSection: Int,
        endSectionKey: String,
        endOffsetInSection: Int
    ) {
        self.startSectionKey = startSectionKey
        self.startOffsetInSection = startOffsetInSection
        self.endSectionKey = endSectionKey
        self.endOffsetInSection = endOffsetInSection
    }
}
